src/hooks/usePastTransaction/usePastTransaction.tsx

```diff
@@ -27,6 +29,25 @@ export const usePastTransaction = (
   const [error, setError] = useState<PastTransactionError | null>(null);
   const prevIds = usePrevious(ids);
   const { selectedIdType } = useContext(IdentificationContext);
+  const [categories, setCategories] = useState<string[]>();
```

Comment:
I noticed that the effect to run `fetchPastTransactions` depends on `categories`, what if we just pass in `categories` as a prop to `usePastTransaction`? So instead of imperatively updating the `categories` state, we can pass it in declaratively.